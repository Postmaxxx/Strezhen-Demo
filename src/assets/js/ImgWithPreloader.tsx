import { useRef, useState } from "react";
import Preloader from "../../components/Preloaders/Preloader";

interface IProps {
	src: string
	alt?: string
	id?: string
}


const ImgWithPreloader: React.FC<IProps> = ({src, alt, id}: IProps):JSX.Element => {

	const [loaded, setLoaded] = useState(false);
	const img = useRef<HTMLImageElement>(null);

	const hasLoaded = () => {
		setLoaded(true)
	}

	return (
		<>
			{loaded || <Preloader />}
			<img ref={img} src={src} alt={alt} onLoad={hasLoaded} style={{display: loaded ? "block" : "none"}} id={id} />
		</>
	);
};


export default ImgWithPreloader;