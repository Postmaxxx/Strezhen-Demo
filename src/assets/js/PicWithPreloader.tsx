import { useEffect, useRef, useState, memo } from "react";
import Preloader from "../../components/Preloaders/Preloader";
import { IImageSizePath, IImageSizes } from "../../interfaces";
 
export type TPartialPathList = {
    [key in keyof IImageSizes]?: string;
};

interface IProps {
	basePath: string
	sizes: IImageSizePath[]
	alt?: string
	id?: string
	image: string
}


const PictureWithPreloader: React.FC<IProps> = ({basePath, sizes = [], alt = '', id, image}: IProps):JSX.Element => {
	
	
	const [loaded, setLoaded] = useState(false);
	const _img = useRef<HTMLImageElement>(null);
	const _spacer = useRef<HTMLDivElement>(null);
	const [containerWidth, setContainerWidth] = useState<number | undefined>(undefined)


	useEffect(() => {
		setContainerWidth(_spacer.current?.clientWidth)
	}, [])


	const hasLoaded = () => {
		setLoaded(true)
	}

	//sizesList is sorted by increasing width
	//search first existed (in pathList) sizename with width for this sizename >= container.width. or just get the last one in the list (the biggest) otherwise
	//const bestSizeName = sizesList?.find(size => size.w >= (containerWidth || 0) && pathList[size.name])?.name || sizesList[sizesList.length - 1].name
	const bestSizePath = sizes.find(size => size.w >= (containerWidth || 0))?.subFolder || sizes[sizes.length - 1].subFolder
	


	return (
		<>
			{loaded || <Preloader />}
			{containerWidth ? 
				<img draggable='false' ref={_img} src={`${basePath}/${bestSizePath}/${image}`} alt={alt} onLoad={hasLoaded} style={{display: loaded ? "block" : "none"}} id={id} />
				: 
				<div ref={_spacer} style={{width: '100%', height: '100%', position: 'absolute', top: '0', left: '0'}}></div> //use mock to calculate the size of container
			}
			
		</>
	);
};

const PicWithPreloader = memo(PictureWithPreloader)
 
export default PicWithPreloader;