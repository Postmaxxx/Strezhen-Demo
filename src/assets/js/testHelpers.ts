import { TLang } from "../../interfaces";
import { screenSizes } from "../../hooks/screenMeter";

type res = 'xs' | 'sm' | 'md' | 'lg' | 'xl'


const setRes = async (res: res, addition: number = 0) => {
    global.innerWidth = screenSizes[res];
    global.innerWidth += addition || 0;
    global.dispatchEvent(new Event('resize'));
}


const changeLang = async (container: HTMLDivElement, act: (callback: () => void | undefined) => void) => {
    let langSwitcher = container.querySelector("[data-testid='lang-switcher']")
    act(() => {
        langSwitcher?.dispatchEvent(new MouseEvent('click', {bubbles: true}));
    });
}

export {setRes, changeLang}