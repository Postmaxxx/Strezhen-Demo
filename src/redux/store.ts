import { createStore, applyMiddleware, combineReducers } from 'redux'
import thunk from 'redux-thunk'
import reducerBase from "./reducers/base";
import reducerNews from "./reducers/news";
import reducerFibers from './reducers/fibers';
import reducerCatalog from './reducers/catalog';
import reducerColors from './reducers/colors';
import reducerUser from './reducers/user';
import reducerContent from './reducers/content';
import reducerOrders from './reducers/orders';
import { composeWithDevTools } from '@redux-devtools/extension';

export const rootReducer = combineReducers({
    base: reducerBase,
    news: reducerNews,
    fibers: reducerFibers,
    catalog: reducerCatalog,
    colors: reducerColors,
    user: reducerUser,
    content: reducerContent,
    orders: reducerOrders
});


const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
 
export default store; 