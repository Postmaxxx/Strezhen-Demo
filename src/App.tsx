import { useEffect, memo, useRef, useState, useMemo  } from "react";
import { Routes, Route, HashRouter } from "react-router-dom";
import { Suspense, lazy } from "react";
import "./assets/css/_base.scss";
import { IFetch, IFullState, TLang } from "./interfaces";
import { AnyAction, bindActionCreators } from "redux";
import { Dispatch } from "redux";
import { connect } from "react-redux";
import P404 from "./pages/P404/P404";
import { allActions } from "./redux/actions/all";
import Footer from "./partials/Footer/Footer";
import Header from "./partials/Header/Header";
import Homer from "./components/Homer/Homer";
import LangSwitcher from "./components/LangSwitcher/LangSwitcher";
import Offliner from "./components/Offliner/Offliner";
import Unauthorized from "./components/Unauthorized/Unauthorized";
import ThemeSwitcher from "./components/ThemeSwitcher/ThemeSwitcher";
import PreloaderPage from "./components/Preloaders/PreloaderPage";
import ModalNew, { IModalFunctions } from "./components/Modal/ModalNew";
import useScreenMeter from "./hooks/screenMeter";
import Preloader from "./components/Preloaders/Preloader";


const LazyLangSwitcher = lazy(() => import("./components/LangSwitcher/LangSwitcher"));
const LazyThemeSwitcher = lazy(() => import("./components/ThemeSwitcher/ThemeSwitcher"));
const LazyHomePage = lazy(() => import("./pages/Home/Home"));
const LazyFibersPage = lazy(() => import("./pages/Fibers/Fibers"));
const LazyFiberPage = lazy(() => import("./pages/Fiber/Fiber"));
const LazyOrderPage = lazy(() => import("./pages/Order/Order"));
const LazyCustomOrderPage = lazy(() => import("./pages/CustomOrder/CustomOrder"));
const LazyOrdersPage = lazy(() => import("./pages/Orders/Orders"));
const LazyCatalogPage = lazy(() => import("./pages/Catalog/Catalog"));
const LazyProduct = lazy(() => import("./pages/Product/Product"));
const LazyNewsDetails = lazy(() => import("./pages/NewsDetails/NewsDetails"));
const LazyFibersCompare= lazy(() => import("./pages/FibersCompare/FibersCompare"));
const LazyNewsCreator = lazy(() => import("./pages/Admin/NewsCreator/NewsCreator"));
const LazyColorCreator = lazy(() => import("./pages/Admin/ColorCreator/ColorCreator"));
const LazyFiberCreator = lazy(() => import("./pages/Admin/FiberCreator/FiberCreator"));
const LazyContactUs = lazy(() => import("./pages/ContactUs/ContactUs"));
const LazyCatalogCahnger = lazy(() => import("./pages/Admin/CatalogCreator/CatalogCreator"));
const LazyProductCreator = lazy(() => import("./pages/Admin/ProductCreator/ProductCreator"));
const LazySpliderChanger = lazy(() => import("./pages/Admin/ContentCreator/contentCreator"));

const ModalMemo = memo(ModalNew)
const FooterMemo = memo(Footer)

interface IPropsState {
    lang: TLang
	isAdmin: boolean
	isAuth: boolean
	fibersLoad: IFetch 
	contentLoad: IFetch
	isLogining: boolean
}

interface IPropsActions {
    setState: {
        fibers: typeof allActions.fibers
        user: typeof allActions.user
		base: typeof allActions.base
		content: typeof allActions.content
    }
}

interface IProps extends IPropsState, IPropsActions {}




const App:React.FC<IProps> = ({lang, isAdmin, isAuth, contentLoad, isLogining, setState}):JSX.Element => {
	const modalRef = useRef<IModalFunctions>(null)

	useEffect(() => {
		setState.user.loginWithToken()
		if (contentLoad.status !== 'success' && contentLoad.status  !== 'fetching') {
			setState.fibers.loadFibers()
		}
	}, [])



	useEffect(() => {
		if (modalRef.current) {
			setState.base.setModal(modalRef)	
		}
	}, [modalRef.current])


    useEffect(() => {
		setState.fibers.loadFibers()
	},[isAdmin])


	const screenWidth = useScreenMeter() 


	return (
		<HashRouter>
			{!screenWidth.sm && <Suspense fallback={<PreloaderPage />}><LazyLangSwitcher /></Suspense>} 
			{!screenWidth.sm && <Suspense fallback={<PreloaderPage />}><LazyThemeSwitcher /></Suspense>} 
			<Homer /> 
			<Offliner lang={lang}/> 
			<Header />
			<Routes>
				<Route index path="/" element={<Suspense fallback={<PreloaderPage />}><LazyHomePage /></Suspense>} />
				
				<Route path="/fibers">
					<Route index element={<Suspense fallback={<PreloaderPage />}><LazyFibersPage /></Suspense>} />
					<Route path="compare" element={<Suspense fallback={<PreloaderPage />}><LazyFibersCompare /></Suspense>} />
					<Route path=":fiberId" element={<Suspense fallback={<PreloaderPage />}><LazyFiberPage /></Suspense>} />
				</Route>
				
				<Route path="/order" element={<Suspense fallback={<PreloaderPage />}>{isAuth ? <LazyOrderPage /> : <Unauthorized lang={lang} />}</Suspense>} />
				<Route path="/custom_order" element={<Suspense fallback={<PreloaderPage />}>{isAuth ? <LazyCustomOrderPage /> : <Unauthorized lang={lang} />}</Suspense>} />
				<Route path="/orders" element={<Suspense fallback={<PreloaderPage />}>{isAuth ? <LazyOrdersPage /> : <Unauthorized lang={lang} />}</Suspense>} />
				<Route path="/contact_us" element={<Suspense fallback={<PreloaderPage />}><LazyContactUs /></Suspense>} />

				<Route path="/catalog">
					<Route index element={<Suspense fallback={<PreloaderPage />}><LazyCatalogPage /></Suspense>} />
					<Route path=":productId" element={<Suspense fallback={<PreloaderPage />}><LazyProduct /></Suspense>} />
				</Route>

				<Route path="news/:newsId" element={<Suspense fallback={<PreloaderPage />}><LazyNewsDetails /></Suspense>} />

				<Route path="/admin">
					<Route path="news-create" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyNewsCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="news-create/:newsId" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyNewsCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="fiber-create" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyFiberCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="fiber-create/:fiberId" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyFiberCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="color-create" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyColorCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="color-create/:colorId" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyColorCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="catalog-change" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyCatalogCahnger /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="product-create" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyProductCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="product-create/:productId" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazyProductCreator /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
					<Route path="splider-change" element={<Suspense fallback={<PreloaderPage />}>{isAdmin ? <LazySpliderChanger /> : !isLogining ? <Unauthorized lang={lang} /> : <Preloader/>}</Suspense>} />
				</Route>

				<Route path="/*" element={<Suspense fallback={<PreloaderPage />}><P404 lang={lang}/></Suspense>} />
			</Routes>
			<FooterMemo lang={lang}/> 
			<ModalMemo ref={modalRef}></ModalMemo>
		</HashRouter>

  );
} 


const mapStateToProps = (state: IFullState): IPropsState => ({
    lang: state.base.lang,
	isAdmin: state.user.isAdmin,
	isAuth: state.user.auth.status === 'success',
	fibersLoad: state.fibers.load,
	contentLoad: state.content.load,
	isLogining: state.user.auth.status === 'fetching',
})

const mapDispatchToProps = (dispatch: Dispatch<AnyAction>):IPropsActions => ({
    setState: {
		fibers: bindActionCreators(allActions.fibers, dispatch),
		user: bindActionCreators(allActions.user, dispatch),
		base: bindActionCreators(allActions.base, dispatch),
		content: bindActionCreators(allActions.content, dispatch),
	}
})
  
      
export default connect(mapStateToProps, mapDispatchToProps)(App)
