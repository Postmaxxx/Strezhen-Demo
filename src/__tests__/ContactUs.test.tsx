import { createRoot } from 'react-dom/client';
import { act, waitFor, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux';
import store from '../redux/store';
import fetchMock from 'jest-fetch-mock';
import App from '../App';
import { setRes } from '../assets/js/testHelpers';
import ContactUs from '../pages/ContactUs/ContactUs';
import renderer from 'react-test-renderer'



describe('ContactUs', () => {
    let _container: HTMLDivElement;
    let _modalContainer: HTMLDivElement;

    beforeEach(() => {
        _container = document.createElement('div');
        _container.id = 'root'
        document.body.appendChild(_container);
        fetchMock.enableMocks()
        fetchMock.doMock()
    });

    afterEach(() => {
        document.body.removeChild(_container);
        fetchMock.resetMocks();
        fetchMock.disableMocks()
    });






    test('should show content', async () => {
        await act(async () => {
            createRoot(_container).render(
                <Provider store={store}>
                    <App />
                </Provider>
            )
        })
        _modalContainer = document.createElement('div');
        _modalContainer.id = 'modal'
        document.body.appendChild(_modalContainer);

        fetchMock.mockResponseOnce(JSON.stringify({ 
            message: {message: {en: 'Message has been sent', ru: 'Сообщение было отправлено'}},  
        }), {
            status: 200,
            headers: { 'content-type': 'application/json' },
        });


        await act(async () => {
            await setRes('sm', 1)
            screen.getByText('КОНТАКТЫ').click()
        })


        let _email: HTMLInputElement = _container.querySelector("#contacter_email") as HTMLInputElement
        let _name: HTMLInputElement = _container.querySelector("#contacter_name") as HTMLInputElement
        let _phone: HTMLInputElement = _container.querySelector("#contacter_phone") as HTMLInputElement
        let _message: HTMLTextAreaElement = _container.querySelector("#contacter_message") as HTMLTextAreaElement
        let _btnSend: HTMLTextAreaElement = screen.getByTestId('contactSendMessage') as HTMLTextAreaElement
        let _map: HTMLImageElement = screen.getByTestId('contactMapImg') as HTMLImageElement
        await waitFor(() => {
            expect(screen.getByTestId('contactHeader')).toBeInTheDocument()
            expect(screen.getByTestId('contactHeader')).toHaveTextContent('Свяжитесь с нами')
            expect(screen.getByTestId('contactSubheader')).toBeInTheDocument()
            expect(screen.getByTestId('contactSubheader')).toHaveTextContent('Компания Стрежень')
            expect(_map).toBeInTheDocument()
            expect(screen.getByTestId('mapLink')).toBeInTheDocument()
            expect((screen.getByTestId('mapLink') as HTMLLinkElement).href.includes('https://go.2gis.com')).toBe(true)
            expect(_email).toBeInTheDocument()
            expect(_name).toBeInTheDocument()
            expect(_phone).toBeInTheDocument()
            expect(_message).toBeInTheDocument()
        })
        
        await act(() => {
            _map.dispatchEvent(new MouseEvent('click', {bubbles: true}))
        })

        await waitFor(async () => {
            let modal = await store.getState().base.modal.current?.getName()
            expect(modal).toBe('location')
        })
        
        await act(() => {
            screen.getByTestId('modal-closer').click()
        })


        await act(() => {
            fireEvent.focus(_email)
            fireEvent.change(_email, { target: { value: 'testmail@gmail.com' } });
            fireEvent.blur(_email)
            fireEvent.focus(_name)
            fireEvent.change(_name, { target: { value: 'TestName' } });
            fireEvent.blur(_name)
            fireEvent.focus(_phone)
            fireEvent.change(_phone, { target: { value: '+322223332' } });
            fireEvent.blur(_phone)
            fireEvent.focus(_message)
            fireEvent.change(_message, { target: { value: 'test_message_with_length_more_than_20' }});
            fireEvent.blur(_message)
            fireEvent.click(_btnSend)
        })


        const sendForm = new FormData()
        sendForm.append('lang', 'ru')
        sendForm.append('message', 'text')


        let lastCall = fetchMock.mock.calls[fetchMock.mock.calls.length - 1]
        expect(lastCall[0]?.toString()).toBe('host/api/user/message')
        expect(lastCall[1]?.method).toBe('POST')
        expect(lastCall[1]?.headers).toEqual({enctype: 'multipart/form-data'})
        
        await waitFor(async () => {
            let modal = await store.getState().base.modal.current?.getName()
            expect(modal).toBe('messageSend')
        })

    })


    test('matches snapshots', async () => {
        let tree = renderer.create(
            <Provider store={store}>
                <ContactUs />
            </Provider>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })


})