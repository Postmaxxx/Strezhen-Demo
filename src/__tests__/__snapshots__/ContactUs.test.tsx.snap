// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs matches snapshots 1`] = `
<div
  className="page_contact"
>
  <div
    className="container_page"
  >
    <div
      className="container"
    >
      <div
        className="block_text"
      >
        <h1
          data-testid="contactHeader"
        >
          Свяжитесь с нами
        </h1>
      </div>
      <form
        className="form_full contact__form"
      >
        <div
          className="block_text"
        >
          <h3
            data-testid="contactSubheader"
          >
            Компания Стрежень
          </h3>
          <div
            className="contacts"
          >
            <div
              className="location"
            >
              <img
                alt="Схема проезда"
                data-testid="contactMapImg"
                onClick={[Function]}
                src="svg-url"
              />
              <a
                data-testid="mapLink"
                href="https://go.2gis.com/uy5m3"
                target="_blank"
              >
                г. Иркутск, Улица Герцена, 14
                 
                <svg
                  className="svg icon_route_2"
                  enableBackground="new 0 0 256 256"
                  version="1.1"
                  viewBox="0 0 256 256"
                  x="0px"
                  xmlSpace="preserve"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                  y="0px"
                >
                  <path
                    d="M162.2,149.3c-1.1-13.9,45.7-13.5,45.7-13.5s-36.4-7.9-53.4,6.2c-21.6,17.9,11,29.3,18.3,36.6c7.3,7.3,8,20.1-24.2,20.9c-32.2,0.7-54.9-1.5-54.9-1.5s-6.4,26.6-12.1,40.6c0,0,61.8,1.8,86.7-4c24.9-5.9,51.2-25.6,39.9-48.3C197,163.6,163.3,163.2,162.2,149.3z"
                  />
                  <path
                    d="M71.5,45.7c-34,0-61.5,27.5-61.5,61.5c0,34,49,60.7,61.5,128.5C89,169,133,141.2,133,107.2C133,73.2,105.4,45.7,71.5,45.7z M71.5,139.8c-18,0-32.6-14.6-32.6-32.6s14.6-32.6,32.6-32.6c18,0,32.6,14.6,32.6,32.6S89.4,139.8,71.5,139.8z"
                  />
                  <path
                    d="M246,53.6c0-20.1-16.3-36.4-36.4-36.4c-20.1,0-36.4,16.3-36.4,36.4c0,20.1,29,36,36.4,76.1C220,90.2,246,73.7,246,53.6z M190.3,53.6c0-10.6,8.6-19.3,19.3-19.3c10.6,0,19.3,8.6,19.3,19.3c0,10.6-8.6,19.3-19.3,19.3C198.9,72.9,190.3,64.2,190.3,53.6z"
                  />
                </svg>
              </a>
            </div>
            <div
              className="contacts__list"
            >
              <div
                className="contact"
              >
                <div
                  className="img-container"
                >
                  <svg
                    className="svg icon_phone"
                    clipRule="evenodd"
                    fillRule="evenodd"
                    height="512px"
                    imageRendering="optimizeQuality"
                    shapeRendering="geometricPrecision"
                    textRendering="geometricPrecision"
                    version="1.1"
                    viewBox="0 0 508 512.06"
                    width="508px"
                    xmlSpace="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlnsXlink="http://www.w3.org/1999/xlink"
                  >
                    <path
                      d="M123.43 249.39c15.07,27.16 32.42,53.26 54.98,77.05 22.61,23.95 50.71,45.72 87.14,64.3 2.7,1.32 5.27,1.32 7.55,0.42 3.46,-1.33 6.99,-4.18 10.47,-7.67 2.7,-2.69 6.04,-6.99 9.56,-11.7 13.99,-18.44 31.34,-41.29 55.81,-29.86 0.54,0.25 0.95,0.53 1.49,0.78l81.66 46.96c0.22,0.12 0.53,0.42 0.77,0.53 10.78,7.41 15.2,18.84 15.32,31.76 0,13.18 -4.85,27.98 -11.96,40.5 -9.4,16.5 -23.27,27.43 -39.25,34.69 -15.2,7 -32.13,10.75 -48.4,13.15 -25.55,3.77 -49.47,1.37 -73.94,-6.16 -23.94,-7.42 -48.03,-19.63 -74.37,-35.89l-1.95 -1.25c-12.07,-7.52 -25.12,-15.61 -37.91,-25.13 -46.91,-35.4 -94.65,-86.48 -125.74,-142.71 -26.08,-47.2 -40.32,-98.15 -32.53,-146.73 4.3,-26.63 15.72,-50.84 35.63,-66.82 17.35,-13.98 40.74,-21.64 71.01,-18.96 3.48,0.23 6.58,2.27 8.19,5.24l52.33 88.5c7.67,9.93 8.61,19.75 4.44,29.58 -3.49,8.09 -10.47,15.5 -20.04,22.46 -2.82,2.4 -6.18,4.84 -9.69,7.4 -11.71,8.5 -25,18.3 -20.45,29.85l-0.12 -0.29zm97.51 -42.25c-8.08,-2.08 -12.94,-10.35 -10.85,-18.43 2.09,-8.08 10.35,-12.94 18.43,-10.85 24.51,6.39 47.56,19.66 66.21,37.07 18.76,17.52 33.31,39.41 40.66,62.88 2.48,7.99 -1.98,16.49 -9.96,18.97 -7.98,2.49 -16.49,-1.97 -18.97,-9.95 -5.75,-18.37 -17.33,-35.69 -32.36,-49.73 -15.15,-14.15 -33.68,-24.88 -53.16,-29.96zm2.01 -92.74c-8.25,-1.4 -13.79,-9.24 -12.39,-17.49 1.39,-8.24 9.24,-13.79 17.48,-12.39 46.21,8 90.08,31.46 125.25,64.38 35.19,32.92 61.82,75.41 73.53,121.46 2.06,8.12 -2.85,16.38 -10.97,18.44 -8.12,2.06 -16.38,-2.85 -18.44,-10.97 -10.3,-40.48 -33.75,-77.89 -64.75,-106.89 -30.93,-28.95 -69.38,-49.55 -109.71,-56.54zm9.74 -84.19c-8.31,-0.95 -14.28,-8.47 -13.33,-16.78 0.95,-8.31 8.46,-14.29 16.77,-13.34 64.65,7.57 125.4,37.22 173.49,80.81 48.21,43.69 83.83,101.51 98.02,165.22 1.8,8.14 -3.35,16.22 -11.5,18.02 -8.14,1.8 -16.22,-3.35 -18.02,-11.5 -12.8,-57.45 -45.05,-109.73 -88.77,-149.34 -43.53,-39.45 -98.4,-66.28 -156.66,-73.09z"
                      fillRule="nonzero"
                    />
                  </svg>
                </div>
                <a
                  href="tel:+79834088949"
                >
                  +7 983 408 89 49
                </a>
              </div>
              <div
                className="contact"
              >
                <div
                  className="img-container"
                >
                  <img
                    alt="VK"
                    src="svg-url"
                  />
                </div>
                <a
                  href="vk.com"
                >
                  Стрежень
                </a>
              </div>
              <div
                className="contact"
              >
                <div
                  className="img-container"
                >
                  <img
                    alt="Telegram"
                    src="svg-url"
                  />
                </div>
                <span>
                  Our group
                </span>
              </div>
              <div
                className="contact"
              >
                <div
                  className="img-container"
                >
                  <svg
                    className="svg icon_email"
                    version="1.1"
                    viewBox="0 0 64 64"
                    x="0px"
                    xmlSpace="preserve"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlnsXlink="http://www.w3.org/1999/xlink"
                    y="0px"
                  >
                    <circle
                      className="st0"
                      cx="32"
                      cy="32"
                      r="32"
                    />
                    <path
                      className="st2"
                      d="M52,44c0,2.2-1.8,4-4,4H16c-2.2,0-4-1.8-4-4V24c0-2.2,1.8-4,4-4h32c2.2,0,4,1.8,4,4V44z"
                    />
                    <path
                      className="st3"
                      d="M52,42c0,2.2-1.8,4-4,4H16c-2.2,0-4-1.8-4-4V22c0-2.2,1.8-4,4-4h32c2.2,0,4,1.8,4,4V42z"
                    />
                    <path
                      className="st2"
                      d="M35.5,30.2c-1.9-2.1-5.1-2.1-7,0L13,43.2c-0.2,0.2-0.3,0.4-0.5,0.6c0.7,1.3,2,2.2,3.4,2.2h32
 c1.5,0,2.7-0.9,3.4-2.2c-0.1-0.2-0.3-0.4-0.5-0.6L35.5,30.2z"
                    />
                    <path
                      className="st3"
                      d="M35.5,32c-1.9-1.9-5.1-1.9-7,0L13,43.5c-0.2,0.2-0.3,0.3-0.5,0.5c0.7,1.2,2,1.9,3.4,1.9h32
 c1.5,0,2.7-0.8,3.4-1.9c-0.1-0.2-0.3-0.3-0.5-0.5L35.5,32z"
                    />
                    <path
                      className="st2"
                      d="M12.6,20.2c0.7-1.3,2-2.2,3.4-2.2h32c1.5,0,2.7,0.9,3.4,2.2c-0.1,0.2-0.3,0.4-0.5,0.6l-15.4,13
 c-1.9,2.1-5.1,2.1-7,0L12.6,20.2z"
                    />
                    <path
                      className="st4"
                      d="M28.5,32c1.9,1.9,5.1,1.9,7,0L51,20.5c0.2-0.2,0.3-0.3,0.5-0.5c-0.7-1.2-2-1.9-3.4-1.9H16
 c-1.5,0-2.7,0.8-3.4,1.9c0.1,0.2,0.3,0.3,0.5,0.5L28.5,32z"
                    />
                  </svg>
                </div>
                <a
                  href="mailto:pypkin@mail.ru"
                >
                  vasya_pypkin@mail.ru
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="block_text"
        >
          <h3>
            Напишите нам сообщение
          </h3>
        </div>
        <div
          className="form__inputs"
        >
          <div
            className="form__inputs__texts"
          >
            <div
              className="block_input"
              data-selector="input-block"
            >
              <label
                htmlFor="contacter_name"
              >
                Ваше имя*
              </label>
              <input
                className="input-element"
                data-selector="input"
                id="contacter_name"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                type="text"
              />
            </div>
            <div
              className="block_input"
              data-selector="input-block"
            >
              <label
                htmlFor="contacter_phone"
              >
                Ваш телефон
              </label>
              <input
                className="input-element"
                data-selector="input"
                id="contacter_phone"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                type="tel"
              />
            </div>
            <div
              className="block_input"
              data-selector="input-block"
            >
              <label
                htmlFor="contacter_email"
              >
                Ваша почта*
              </label>
              <input
                className="input-element"
                data-selector="input"
                id="contacter_email"
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                type="email"
              />
            </div>
            <div
              className="block_input expandable"
              data-selector="input-block"
            >
              <label
                htmlFor="contacter_message"
              >
                Ваше сообщение*
              </label>
              <textarea
                className="input-element"
                data-selector="input"
                id="contacter_message"
                onBlur={[Function]}
                onChange={[Function]}
              />
            </div>
          </div>
          <div
            className="form__inputs__files"
          >
            <div
              className="block_input files"
            >
              <label
                htmlFor="files"
              >
                Прикрепить файлы
              </label>
              <div
                className="files-adder"
              >
                <svg
                  className="files-adder__arrow"
                  height="25"
                  viewBox="0 0 16 25"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.31719 23.7322C7.71856 24.1116 8.35148 24.0937 8.73084 23.6924L14.9129 17.1516C15.2923 16.7502 15.2745 16.1173 14.8731 15.7379C14.4717 15.3586 13.8388 15.3764 13.4594 15.7778L7.96424 21.5918L2.15022 16.0966C1.74885 15.7172 1.11593 15.7351 0.73657 16.1365C0.357206 16.5378 0.375048 17.1707 0.776422 17.5501L7.31719 23.7322ZM6.36655 0.404461L7.00449 23.0336L9.00369 22.9773L8.36576 0.348102L6.36655 0.404461Z"
                  />
                </svg>
                <div
                  className="files-adder__label"
                >
                  <label
                    htmlFor="files"
                  >
                    Выберите файлы
                  </label>
                  <span>
                    для добавления или перетащите их сюда
                  </span>
                </div>
                <input
                  className="button_add-files"
                  id="files"
                  multiple={true}
                  onChange={[Function]}
                  type="file"
                />
                <div
                  className="files-adder__previewer"
                />
              </div>
            </div>
          </div>
        </div>
        <button
          className="button_blue button_contact"
          data-testid="contactSendMessage"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Отправить сообщение
        </button>
      </form>
    </div>
  </div>
</div>
`;
